handlers/search.go

```diff
@@ -127,153 +123,117 @@ func SearchAhead(w http.ResponseWriter, r *http.Request) {
 			GROUP BY graffiti
 			ORDER BY count desc
 			LIMIT 10`, "%"+search+"%")
-		if err == nil {
-			for i := range *graffiti {
-				(*graffiti)[i].Graffiti = utils.FormatGraffitiString((*graffiti)[i].Graffiti)
-			}
+		if err != nil {
+			break
+		}
+		for i := range *graffiti {
+			(*graffiti)[i].Graffiti = utils.FormatGraffitiString((*graffiti)[i].Graffiti)
 		}
 		result = graffiti
 	case "transactions":
+		if !transactionLikeRE.MatchString(strings.ToLower(strings.Replace(search, "0x", "", -1))) {
+			break
+		}
 		result = &types.SearchAheadTransactionsResult{}
-		if transactionLikeRE.MatchString(strings.ToLower(strings.Replace(search, "0x", "", -1))) {
-			txHash, txHashErr := hex.DecodeString(strings.ToLower(strings.Replace(search, "0x", "", -1)))
-			if txHashErr != nil {
-				logger.Errorf("error parsing txHash %v: %v", search, txHashErr)
-				http.Error(w, "Internal server error", http.StatusInternalServerError)
-				return
-			}
-			var tx *types.Eth1TransactionIndexed
-			tx, err = db.BigtableClient.GetIndexedEth1Transaction(txHash)
-			if err == nil && tx != nil {
-				result = &types.SearchAheadTransactionsResult{{TxHash: fmt.Sprintf("%x", tx.Hash)}}
-			}
+		var txHash []byte
+		txHash, err = hex.DecodeString(strings.ToLower(strings.Replace(search, "0x", "", -1)))
+		if err != nil {
+			err = fmt.Errorf("error parsing txHash %v: %v", search, err)
+			break
+		}
+		var tx *types.Eth1TransactionIndexed
+		tx, err = db.BigtableClient.GetIndexedEth1Transaction(txHash)
+		if err == nil && tx != nil {
+			result = &types.SearchAheadTransactionsResult{{TxHash: fmt.Sprintf("%x", tx.Hash)}}
 		}
 	case "epochs":
+		var epochNumber uint64
+		epochNumber, err = strconv.ParseUint(search, 10, 32)
```

Comment:
This introduces lots of errors in our log file @recy21 will not be thrilled about. Search for "X" and the "epochs" search type will log something like
`level=error msg="error doing query for searchAhead" error="strconv.ParseUint: parsing \"X\": invalid syntax" module=handlers searchType=epochs`

The same applies here: https://github.com/gobitfly/eth2-beaconchain-explorer/blob/76e74910164abad2e678b45b693ce2ddaa81b913/handlers/search.go#L105

Please handle conversion errors like done for slots here:
https://github.com/gobitfly/eth2-beaconchain-explorer/blob/76e74910164abad2e678b45b693ce2ddaa81b913/handlers/search.go#L83

Reply:
should be fixed, ptal